authappliance/menu.py

```diff
@@ -301,6 +303,140 @@ def _execute_remote_sql(self, sql):
+        pinode2_filename = os.path.join(tinc_net_directory, 'hosts', 'pinode2')
+        local_pinode1_conf = TincConfFile(local_io_handler, pinode1_filename)
+        local_pinode1_conf.update({
+            'Address': str(self.local_ip),
+            'Subnet': local_vpn_ip,
+        })
+        local_pinode1_conf.save()
+        # Remote configure the pinode2 host file (which already contains the pubkey)
+        remote_pinode2_conf = TincConfFile(remote_io_handler, pinode2_filename)
+        remote_pinode2_conf.update({
+            'Address': str(self.remote_ip),
+            'Subnet': remote_vpn_ip,
+        })
+        remote_pinode2_conf.save()
+
+        # Exchange host files (and thus RSA pubkeys)
+        # pinode1 -> REMOTE
+        sftp.put(pinode1_filename, pinode1_filename)
+        # REMOTE -> pinode2
+        sftp.get(pinode2_filename, pinode2_filename)
+
+        # Create tinc-up scripts
+        tinc_up_filename = os.path.join(tinc_net_directory, 'tinc-up')
+        tinc_up_config = '\n'.join([
+            'ip link set $INTERFACE up',
+            'ip addr add {ip} dev $INTERFACE',
+            'ip route add {network} dev $INTERFACE'])
+
+        # locally
+        local_tinc_up = UpScript(local_io_handler, tinc_up_filename)
+        local_tinc_up.appliance_section = tinc_up_config.format(
+            ip=local_vpn_ip,
+            network=vpn_subnet,
+        ).split('\n')
+        local_tinc_up.save()
+        local_io_handler.chmod(tinc_up_filename, 0o755)
+
+        # remotely
+        remote_tinc_up = UpScript(remote_io_handler, tinc_up_filename)
+        remote_tinc_up.appliance_section = tinc_up_config.format(
+            ip=remote_vpn_ip,
+            network=vpn_subnet,
+        ).split('\n')
+        remote_tinc_up.save()
+        remote_io_handler.chmod(tinc_up_filename, 0o755)
+
+        # add network to nets.boot
+        nets_boot_filename = '/etc/tinc/nets.boot'
+        # locally
+        local_nets_boot = NetsBoot(local_io_handler, nets_boot_filename)
+        local_nets_boot.add(vpn_name)
+        local_nets_boot.save()
+
+        # remotely
+        remote_nets_boot = NetsBoot(remote_io_handler, nets_boot_filename)
+        remote_nets_boot.add(vpn_name)
+        remote_nets_boot.save()
+
+        # Start the tinc nets
+        start_command = 'tincd -n {}'.format(pipes.quote(vpn_name))
+        # locally
+        proc = Popen(start_command, shell=True)
+        if proc.wait() != 0:
+            self.add_info('Could not bring up the tinc VPN locally!')
+        # remotely
+        stdin, stdout, stderr = ssh.exec_command(start_command)
+        stdout.channel.recv_exit_status()
+        if stderr:
+            self.add_info(stderr.read())
+
+        # and we are done.
+        # we now set local_ip and remote_ip to the VPN IP addresses!
+        self.local_ip = local_vpn_ip
+        self.remote_ip = remote_vpn_ip
+
+        self.add_info('The tinc VPN was set up successfully!')
```

Comment:
The tinc daemon usually starts even if it can not setup the vpn.
So before printing "successfully" we probably should somehow test the connect maybe with a ping to the remove VPN address.

Reply:
added this in 8c94949, but also had to add a 1-second delay before trying to ping